### PR TITLE
chore(dev): initialize databases kong and kong_tests

### PR DESCRIPTION
**Pull request by outsinre**
### Summary

`POSTGRES_DBS` is not supported by official Postgres Docker image at https://github.com/Kong/kong/blob/4e203297c394e870b25122c5b3cb24e4135ae391/scripts/dependency_services/docker-compose-test-services.yml#L10

We have to initialize the database explicitly. It is verified on both macOS `arm64` and Linux `x86_64`.

Before the fix:

```bash
root@b6d67e271744:/# psql -U kong
psql (15.2 (Debian 15.2-1.pgdg110+1))
Type "help" for help.

kong=# \l kong_tests
                                      List of databases
 Name | Owner | Encoding | Collate | Ctype | ICU Locale | Locale Provider | Access privileges
------+-------+----------+---------+-------+------------+-----------------+-------------------
(0 rows)

kong=#
```

After the fix:

```bash
16:52:27 zachary@Zacharys-MacBook-Pro ~/gk/kong-master$ docker compose -f scripts/dependency_services/docker-compose-test-services.yml up -d
[+] Running 11/11
 ✔ Network dependency_services_default                                                                                                                    Created                                                                                                          0.0s
 ✔ Volume "dependency_services_cassandra-data"                                                                                                            Created                                                                                                          0.0s
 ✔ Volume "dependency_services_redis-data"                                                                                                                Created                                                                                                          0.0s
 ✔ Volume "dependency_services_postgres-data"                                                                                                             Created                                                                                                          0.0s
 ✔ Container dependency_services-redis-1                                                                                                                  Started                                                                                                          0.9s
 ✔ Container dependency_services-zipkin-1                                                                                                                 Started                                                                                                          0.8s
 ✔ Container dependency_services-grpcbin-1                                                                                                                Started                                                                                                          1.0s
 ✔ Container dependency_services-postgres-1                                                                                                               Started                                                                                                          0.8s
 ✔ Container dependency_services-cassandra-1                                                                                                              Started                                                                                                          1.0s
 ! zipkin The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested                                                                                                                   0.0s
 ! grpcbin The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested                                                                                                                  0.0s
16:52:44 zachary@Zacharys-MacBook-Pro ~/gk/kong-master$ docker compose ls
NAME                  STATUS              CONFIG FILES
dependency_services   running(5)          /Users/zachary/gk/kong-master/scripts/dependency_services/docker-compose-test-services.yml
kong                  running(5)          /Users/zachary/workspace/kong-ee/scripts/dependency_services/docker-compose-test-services.yml
16:52:50 zachary@Zacharys-MacBook-Pro ~/gk/kong-master$ docker compose -p dependency_services ps
NAME                              IMAGE                    COMMAND                  SERVICE             CREATED             STATUS                             PORTS
dependency_services-cassandra-1   cassandra:3              "docker-entrypoint.s…"   cassandra           13 seconds ago      Up 11 seconds (health: starting)   7001/tcp, 127.0.0.1:60730->7000/tcp, 127.0.0.1:60731->7199/tcp, 127.0.0.1:60732->9042/tcp, 127.0.0.1:60733->9160/tcp
dependency_services-grpcbin-1     moul/grpcbin             "/bin/grpcbin"           grpcbin             13 seconds ago      Up 11 seconds                      80/tcp, 127.0.0.1:60734->9000/tcp, 127.0.0.1:60735->9001/tcp
dependency_services-postgres-1    postgres                 "docker-entrypoint.s…"   postgres            13 seconds ago      Up 11 seconds (healthy)            127.0.0.1:60727->5432/tcp
dependency_services-redis-1       redis                    "docker-entrypoint.s…"   redis               13 seconds ago      Up 11 seconds (healthy)            127.0.0.1:60728->6379/tcp, 127.0.0.1:60729->6380/tcp
dependency_services-zipkin-1      openzipkin/zipkin:2.19   "/busybox/sh run.sh"     zipkin              13 seconds ago      Up 11 seconds                      9410/tcp, 127.0.0.1:60726->9411/tcp
16:52:56 zachary@Zacharys-MacBook-Pro ~/gk/kong-master$ docker compose -p dependency_services exec -it postgres bash
root@55d2dcd7c670:/# psql -U kong
psql (15.2 (Debian 15.2-1.pgdg110+1))
Type "help" for help.

kong-# \l
                                             List of databases
    Name    | Owner | Encoding |  Collate   |   Ctype    | ICU Locale | Locale Provider | Access privileges
------------+-------+----------+------------+------------+------------+-----------------+-------------------
 kong       | kong  | UTF8     | en_US.utf8 | en_US.utf8 |            | libc            | =Tc/kong         +
            |       |          |            |            |            |                 | kong=CTc/kong
 kong_tests | kong  | UTF8     | en_US.utf8 | en_US.utf8 |            | libc            | =Tc/kong         +
            |       |          |            |            |            |                 | kong=CTc/kong
 postgres   | kong  | UTF8     | en_US.utf8 | en_US.utf8 |            | libc            |
 template0  | kong  | UTF8     | en_US.utf8 | en_US.utf8 |            | libc            | =c/kong          +
            |       |          |            |            |            |                 | kong=CTc/kong
 template1  | kong  | UTF8     | en_US.utf8 | en_US.utf8 |            | libc            | =c/kong          +
            |       |          |            |            |            |                 | kong=CTc/kong
(5 rows)
```

### Checklist

- [n/a] The Pull Request has tests
- [n/a] There's an entry in the CHANGELOG
- [n/a] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Full changelog

* Add an initialization script to `/docker-entrypoint-initdb.d`.
* Fix a shell array reference bug.
* make sure `rocks_config` and `resty` can be overwritten

### Issue reference

Fix #_[FTI-5013]_

[FTI-5013]: https://konghq.atlassian.net/browse/FTI-5013?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by hanshuebner — COMMENTED** [review]
Would it make sense to guard against the existence of the databases, or is it certain that they don't exist in the mounted host directory when the docker-compose file is run?

**Comment by dndx** [comment]
@outsinre @fffonion CE does not need this fix?

**Comment by fffonion** [comment]
yes, this script should be made idempotent: meaning rerunning the script muiltiple times won't cause an error.
@outsinre please also open a PR on CE side.

**Comment by outsinre** [comment]
> Would it make sense to guard against the existence of the databases, or is it certain that they don't exist in the mounted host directory when the docker-compose file is run?

Make sense, though it is rare that the database already exists. It is rare as the script is bound to **initialization phase** of the Postgres service, namely the very first step after the docker is spawned.

Let me try to dd a restriction.

**Comment by outsinre** [comment]
> 

@fffonion, @dndx this is the PR on CE side. Will cherry-pick to EE side after merge.

**Comment by outsinre** [comment]
> yes, this script should be made idempotent: meaning rerunning the script muiltiple times won't cause an error. @outsinre please also open a PR on CE side.

Make sense, in case the script is manually called.

File: scripts/dependency_services/00-create-pg-db.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+shared_preload_libraries = 'pg_stat_statements'
+track_activity_query_size = 2048
+pg_stat_statements.track = all
+pg_stat_statements.max = 10000
+" >>$pg_conf_file
+
+for database in $(echo $POSTGRES_DBS | tr ',' ' '); do
+  echo "Creating database $database"
+  psql -U $POSTGRES_USER <<-EOSQL
+    SELECT 'CREATE DATABASE $database' WHERE NOT EXISTS (SELECT FROM pg_database WHERE datname = '$database')\gexec
```
Comment (hanshuebner) [inline scripts/dependency_services/00-create-pg-db.sh:19]: TIL `\gexec`, nice!

**Review by hanshuebner — APPROVED** [review]

**Comment by outsinre** [comment]
Will cherry pick to EE.

**Comment by outsinre** [comment]
Cherry-pick: https://github.com/Kong/kong-ee/pull/5082

**Pull request merged**